crates/sui-indexer/migrations_v2/2023-08-19-044035_tx_indices/up.sql

```diff
@@ -0,0 +1,19 @@
+CREATE TABLE tx_indices (
+    tx_sequence_number          BIGINT       PRIMARY KEY,
+    transaction_digest          bytea        NOT NULL,
+    input_objects               bytea[]      NOT NULL,
+    changed_objects             bytea[]      NOT NULL,
+    senders                     bytea[]      NOT NULL,
+    recipients                  bytea[]      NOT NULL,
+    packages                    bytea[]      NOT NULL,
+    package_modules             text[]       NOT NULL,
+    package_module_functions    text[]       NOT NULL
+);
+
+CREATE INDEX tx_indices_input_objects ON tx_indices USING GIN(input_objects);
```

Comment:
How does a GIN on a byte-array work? Are we effectively indexing all sub-arrays here, or is it doing something cleverer to index each Object ID?

Reply:
>  is it doing something cleverer to index each Object ID?

yes https://www.postgresql.org/docs/current/gin-intro.html

Reply:
I think the detail I was missing here is that `bytea` is already an array (i.e. I was worried that `input_objects` was an array of bytes, rather than an array of byte arrays, each being an ObjectID).